control/controlclient/direct_test.go

```diff
@@ -103,3 +105,46 @@ func TestNewHostinfo(t *testing.T) {
 	}
 	t.Logf("Got: %s", j)
 }
+
+func TestTsmpPing(t *testing.T) {
+	hi := NewHostinfo()
+	ni := tailcfg.NetInfo{LinkType: "wired"}
+	hi.NetInfo = &ni
+
+	key, err := wgkey.NewPrivate()
+	if err != nil {
+		t.Error(err)
+	}
+	opts := Options{
+		ServerURL: "https://example.com",
+		Hostinfo:  hi,
+		GetMachinePrivateKey: func() (wgkey.Private, error) {
+			return key, nil
+		},
+	}
+
+	c, err := NewDirect(opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	now := time.Now()
+	res := &ipnstate.PingResult{
+		IP:       "123.456.7890",
+		Err:      "",
+		NodeName: "testnode",
+	}
+
+	pr := &tailcfg.PingRequest{
+		URL: "testurl.com",
```

Comment:
Is this going to send a HTTP request to `testurl.com`?

I'd recommend something under `example.com` because that's guaranteed to always be reserved. But also, I think the `httpc` should be a special test http client that doesn't actually make real requests but instead lets you inspect the contents. Or perhaps set the URL to a localhost HTTP server set up by the test that captures the result to inspect.

Reply:
now using httptest server